ape_foundry/provider.py

```diff
@@ -385,6 +385,9 @@ def set_balance(self, account: AddressType, amount: Union[int, float, str, bytes
     def set_timestamp(self, new_timestamp: int):
         self._make_request("evm_setNextBlockTimestamp", [new_timestamp])
 
+    def get_auto_mine(self) -> bool:
+        return self._make_request("anvil_getAutomine", [])
```

Comment:
just to be consistent with other read-only properties, this should probably be a property too
```suggestion
    @property
    def auto_mine(self) -> bool:
        return self._make_request("anvil_getAutomine", [])
```

Reply:
you gotta be careful with properties that makes requests under-the-hood:

```python
if self.auto_mine:
   # do things that cause self.auto_mine to change.
   cached_val = self.auto_mine # was expecting true but it changed to false
```

Reply:
fair, but also this is why we have the walrus operator :wink: 
```py
if cached_val := self.auto_mine:
   # do things that cause self.auto_mine to change.
```

Reply:
ya you can set it to a variable, that is all that does, that is like a workaround tho

Reply:
but actually... im ok with it being a property... i dont see it changing too often or anything